src/DistributedArrays.jl

```diff
@@ -1253,4 +1256,84 @@ function Ac_mul_B(A::DMatrix, B::AbstractMatrix)
     return Ac_mul_B!(one(T), A, B, zero(T), C)
 end
 
+
+# Sorting a DVector using samplesort
+
+function sample_n_setup_ref(d::DVector, sample_size)
+    lp = localpart(d)
+    llp = length(lp)
+    sample_size = llp > sample_size ? sample_size : llp
+    sample = lp[rand(1:llp, sample_size)]
+    ref = RemoteChannel(()->Channel(length(procs(d))))             # To collect parts to be sorted locally later.
+    return (sample, ref)
+end
+
+
+function scatter_n_sort_localparts{T}(d::DVector{T}, refs::Array{RemoteChannel}, boundaries::Array{T})
+    # send respective parts to correct workers
+    lp=localpart(d)
+    for (i,p) in enumerate(procs(d))
+        part_on_worker_i=lp[find(x-> (x >= boundaries[i]) && (x < boundaries[i+1]), lp)]
+        @async put!(refs[i], part_on_worker_i)
+    end
+
+    myidx = findfirst(procs(d), myid())
+
+    # wait to receive all of my parts from all other workers
+    lp_sorting=T[]
+    for _ in procs(d)
+        append!(lp_sorting, take!(refs[myidx]))
+    end
+
+    sorted_ref=RemoteChannel()
+    put!(sorted_ref, sort!(lp_sorting))
+
+    return sorted_ref
+end
+
+
+function samplesort{T}(d::DVector{T})
```

Comment:
Maybe follow the style for `sort` in base where we dispatch on an algorithm type.